sdks/python/apache_beam/io/external/jdbc.py

```diff
@@ -0,0 +1,134 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+  PTransforms for supporting Jdbc in Python pipelines. These transforms do not
+  run a Jdbc client in Python. Instead, they expand to ExternalTransforms
+  which the Expansion Service resolves to the Java SDK's JdbcIO.
+
+  Note: To use these transforms, you need to start a Java Expansion Service.
+  Please refer to the portability documentation on how to do that. Flink Users
+  can use the built-in Expansion Service of the Flink Runner's Job Server. The
+  expansion service address has to be provided when instantiating the
+  transforms.
+
+  If you start Flink's Job Server, the expansion service will be started on
+  port 8097. This is also the configured default for this transform. For a
+  different address, please set the expansion_service parameter.
+
+  For more information see:
+  - https://beam.apache.org/documentation/runners/flink/
+  - https://beam.apache.org/roadmap/portability/
+"""
+
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import typing
+
+from past.builtins import unicode
+
+from apache_beam.transforms.external import BeamJarExpansionService
+from apache_beam.transforms.external import ExternalTransform
+from apache_beam.transforms.external import NamedTupleBasedPayloadBuilder
+
+__all__ = ['WriteToJdbc']
+
+
+def default_io_expansion_service():
+  return BeamJarExpansionService('sdks:java:io:expansion-service:shadowJar')
+
+
+WriteToJdbcSchema = typing.NamedTuple(
+    'WriteToJdbcSchema',
+    [
+        ('driver_class_name', unicode),
+        ('jdbc_url', unicode),
+        ('username', unicode),
+        ('password', unicode),
+        ('statement', unicode),
+        ('connection_properties', unicode),
+        ('connection_init_sqls', typing.Optional[typing.List[unicode]]),
+    ],
+)
+
+
+class WriteToJdbc(ExternalTransform):
+  """
+  An external PTransform which writes Rows to the specified database.
+
+  This transform receives Rows defined as NamedTuple type and registered in
+  the coders registry to use RowCoder, e.g.
+
+    import typing
+    from apache_beam import coders
+
+    ExampleRow = typing.NamedTuple(
+      "ExampleRow",
+      [
+        ("id", int),
+        ("name", unicode),
+        ("budget", float),
+      ],
+    )
+    coders.registry.register_coder(ExampleRow, coders.RowCoder)
+
+  Experimental; no backwards compatibility guarantees.  It requires special
+  preparation of the Java SDK.  See BEAM-7870.
+  """
+
+  URN = 'beam:external:java:jdbc:write:v1'
+
+  def __init__(
+      self,
+      driver_class_name,
+      jdbc_url,
+      username,
+      password,
+      statement,
+      connection_properties='',
+      connection_init_sqls=None,
+      expansion_service=None,
+  ):
```

Comment:
Could you make these parameters into keyword args? It's easy for a user to get lost in a long list of function parameters, but the keywords args will make it explicit. The only cost is you'll need explicit checks the the required parameters aren't `None`.

Reply:
Aren't kwargs more confusing to the user and forces him to look into documentation? How about leaving the required params as they are and use kwargs for the optional ones?
For me it's always useful that IDE tells me what params are yet to pass.
I'm more a Java than Python guy so I may be wrong.

Reply:
Hm so I was under the impression that in Python positional arguments had to be passed as positional args, I didn't realize that callers are allowed to name them as you've done in the test:
```python
          | 'Write to jdbc' >> WriteToJdbc(
              driver_class_name=self.driver_class_name,
              jdbc_url=self.jdbc_url,
              username=self.username,
              password=self.password,
              statement='INSERT INTO {} VALUES(?, ?, ?)'.format(table_name),
          ))
```
So thanks for teaching me something new! :)

My concern was that users would end up passing a long parameter list of all primitive types, which can be hard to keep track of (e.g. in Java you'd use a Builder rather than having a long parameter list like that). But since users can just name the args as you've done that alleviates my concern.

Reply:
Oh ok, I thought that you meant using **kwargs. It confused me but since I'm not familiar with python tricks I thought that there is an advantage of using it. But if you only meant passing named parameters then I'll switch back to the default values as then a user has IDE suggestions.